int32 x
int32 y
int32 z
int32 thta
int32 walking_mode
int32 walking_state
